ironic/drivers/modules/drac/raid.py

```diff
@@ -293,6 +293,43 @@ def clear_foreign_config(node, raid_controller):
                    'error': exc})
         raise exception.DracOperationError(error=exc)
 
+def change_physical_disk_state(node, mode,
+                               controllers_to_physical_disk_ids=None):
+    """
+    :param node: an ironic node objec
+    :param mode: enumeration that indicates the mode
+                     to change the disks to.
+    :param controllers_to_physical_disk_ids:Dictionary of controllers and
+                                            corresponding disk ids to convert
+                                            to the requested mode
+    :return: a dictionary containing:
+             - conversion_results, a dictionary that maps controller ids
+             to the conversion results for that controller.  The
+             conversion results are a dict that contains:
+             - The is_commit_required key with the value always set to
+             True indicating that a config job must be created to
+             complete disk conversion.
+             - The is_reboot_required key with a RebootRequired
+             enumerated value indicating whether the server must be
+             rebooted to complete disk conversion.
+    """
+    try:
+
+        drac_job.validate_job_queue(node)
+
+        client = drac_common.get_drac_client(node)
+        return client.change_physical_disk_state(
+            mode,
+            controllers_to_physical_disk_ids=controllers_to_physical_disk_ids)
+    except drac_exceptions.BaseClientException as exc:
+        LOG.error('DRAC driver failed to free foreign driver '
```

Comment:
This error message references freeing foreign drives.  Looks like it  needs to be updated.

Reply:
Done